numpy/core/src/umath/ufunc_object.c

```diff
@@ -1025,6 +1029,13 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
             }
 
             switch (str[0]) {
+                case 'a':
```

Comment:
If speed really matters here, then presumably we could save a little more with `strcmp(str+1, "xis")` on the following lines...

Either way, I think this probably needs a comment explaining that we're avoiding PyArg_ParseTupleAndKeywords for speed reasons.